airbyte-api/src/main/openapi/config.yaml

```diff
@@ -317,6 +322,66 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/SourceDefinitionReadList"
+  /v1/source_definitions/list_opt_in:
+    post:
+      tags:
+        - source_definition
+      summary: List all opt-in sourceDefinitions along with the current workspace's opt-in statuses
+      operationId: listSourceDefinitionOptIns
```

Comment:
I think this would read more naturally as `listOptInSourceDefinitions` (or `listPrivateSourceDefinitions` if we go that route (see above comment))